### PR TITLE
Fix flaky test in recoverseg_from_file

### PR DESCRIPTION
**Pull request by zhangh43**
1. after stop primary with content=1, we should check promotion
    status with 1U:
2. after manually update dbid, we should trigger a fts probe and
    wait for the mirror promotion as well.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql [hunk 1]
```diff
@@ -34,21 +34,30 @@ where c.role='p' and c.content=1), 'stop');
...
+-- wait for content 1 (earlier mirror, now primary) to finish the promotion
+1U: select 1;
 -- Quit this utility mode session, as need to start fresh one below
-0Uq:
+1Uq:
 
 -- make the dbid in gp_segment_configuration not continuous
+-- dbid=2 corresponds to content id =0
 set allow_system_table_mods to true;
 update gp_segment_configuration set dbid=9 where dbid=2;
```
Comment (paul-guo-) [inline src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql:45]: Interesting, dbid change in gp_segment_configuration will cause fts mark down. Can we use content=0 and role='p' instead. So no need of comment of "dbid=2 corresponds to content id =0". Also mention that dbid 9 is a dbid number that is not used in the test demo cluster.

File: src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql [hunk 2]
```diff
@@ -62,13 +71,16 @@ select dbid from gp_segment_configuration where dbid=2;
 update gp_segment_configuration set dbid=2 where dbid=9;
 set allow_system_table_mods to false;
 
--- we manually change dbid from 2 to 9, which casue the
+-- we manually change dbid from 2 to 9, which causes the
 -- corresponding segment down as well, so recovery full
 -- at here
```
Comment (paul-guo-) [inline src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql:76]: Doesn't incremental recovery work? Full recovery is so slow if the node has many data.

Comment (zhangh43) [reply src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql:76]: Previous `gprecoverseg -i` will just recover the selected segment. But I think `gprecoverseg` would work here.  I'll try it.

File: src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql (hunk 1 above)
Comment (zhangh43) [reply src/test/isolation2/sql/segwalrep/recoverseg_from_file.sql:45]: Yes, will update it.

**Comment by zhangh43** [comment]
Merged. Thanks for the review!

**Pull request merged**